### DISCUSSION TITLE
Simplify the simpliest set / setter `setFoo: foo => set({ foo })` ?

### DISCUSSION THREAD

**Discussion by kkocdko** [pmndrs/zustand#1706] [Q&A] [answered]
Example in `README.md` shows a `increase()` method, it's great.

```js
const useBearStore = create((set) => ({
  bears: 0,
  increase: () => set((state) => ({ bears: state.bears + 1 }))
}));
```

However in many case I only need the simpliest setter:

```js
const useStore = create((set) => ({
  colorScheme: "auto" ,
  setColorScheme: (v) => set({ colorScheme: v }),
  autoUpdate: "enabled",
  setAutoUpdate: (v) => set({ autoUpdate: v }),
  /// and many many states ...
}));
```

Then in my app there's so many state like this, is there any way to avoid define `setFoo: foo => set({ foo })` manually?

(I searched exists issues and discussions but found nothing, probably the wrong keywords, say sorry if there are duplicates)

**Comment by dai-shi** [comment, marked as answer]
We do have auto generating selectors in docs, but not auto generating actions.
You could do it easily with JS, but TS typing will be a challenge.

**Reply by kkocdko** [reply]
Thanks for your answer.

And [auto-generating-selectors docs link here](https://github.com/pmndrs/zustand/blob/main/docs/guides/auto-generating-selectors.md) for who want it.

**Reply by JacobWeisenburger** [reply]
I'm not really sure how that link solves this issue. Can you explain more?
